### PR TITLE
[PUB-2129] Refactor user tagging modal

### PR DESCRIPTION
**Pull request by hamstu**
## Description

This PR refactors the user tagging modal into its own component that is then brought into the composer.

## Context & Notes

- This is still WIP, the modal is off-center and needs some other style tweaks. This will happen in a future PR.
- Jira: https://buffer.atlassian.net/browse/PUB-2129

## Screenshots (if appropriate)

image

## Checklist

-   [x] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have tested different user stories. 
-   [ ] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/composer-user-tags/components/story.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { withA11y } from '@storybook/addon-a11y';
+// import { action } from '@storybook/addon-actions';
+
+import UserTags from './index';
+
+storiesOf('Composer User Tags', module)
```
Comment (albennett) [inline packages/composer-user-tags/components/story.jsx]: 💯

File: packages/composer/composer/components/InstagramUserTags.jsx [hunk 2]
```diff
@@ -1,222 +1,23 @@
...
-  background-color: white;
-  padding: 0px 15px 0px 0px;
-  display: flex;
-`;
-
-const StyledRightHeader = styled.div`
-  display: flex;
-  align-items: center;
-`;
+import UserTags from '../../../composer-user-tags';
```
Comment (albennett) [inline packages/composer/composer/components/InstagramUserTags.jsx]: Maybe we can add a direct import `@bufferapp/publish-composer-user-tags`

File: packages/composer/composer/components/InstagramUserTags.jsx [hunk 3]
```diff
@@ -1,222 +1,23 @@
...
-            Note: The Instagram API only allows public users to be
-            tagged and prevents autocomplete in usernames. // Move to translations
-          </StyledText>
-          <Button
-            onClick={saveGlobalTags}
-            label="Save and Close"
-          />
-        </StyledRightContent>
-      </StyledModalWrapper>
+    <Modal hideCloseButton>
```
Comment (albennett) [inline packages/composer/composer/components/InstagramUserTags.jsx]: Does the modal still look ok without the classNames?

Comment (hamstu) [reply packages/composer/composer/components/InstagramUserTags.jsx]: It still works but it's off-center a bit. I was planning to refactor / fix this in my next PR. 😄

File: packages/composer/composer/components/InstagramUserTags.jsx (hunk 2 above)
Comment (hamstu) [reply packages/composer/composer/components/InstagramUserTags.jsx]: Yeah! Much better that way. I'll fix it up 👍

**Pull request merged**